src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/IVsSolutionRestoreService.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.SolutionRestoreManager
+{
+    /// <summary>
+    /// Represents a package restore service API for integration with a project system.
+    /// </summary>
+    [ComImport]
+    [Guid("2b046428-ca39-40bb-8b4b-7dd1d96118cb")]
+    public interface IVsSolutionRestoreService
+    {
+        /// <summary>
+        /// A task providing last/current restore operation status. 
+        /// If the restore is in-progress task will be not completed.
+        /// CPS will rely on the task completion status to block solution build.
+        /// </summary>
+        Task<bool> CurrentRestoreOperation { get; }
+
+        /// <summary>
+        /// An entry point used by CPS to indicate given project needs to be restored.
+        /// </summary>
+        /// <param name="projectUniqueName">
+        /// Unique identificator of the project. Should be a full path to project file.
+        /// </param>
+        /// <param name="projectRestoreInfo">Metadata <see cref="IVsProjectRestoreInfo"/> needed for restoring the project.</param>
+        /// <param name="token">Cancellation token.</param>
+        /// <returns>
+        /// Returns if the requested nuget restore operation for the given project 
+        /// was a success or failure.
+        /// </returns>
+        Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectRestoreInfo projectRestoreInfo, CancellationToken token);
```

Comment:
No implementation yet (WIP). Correct, `IVsHierarchy` was eliminated from the API.